pkg/obfuscator/tracker.go

```diff
@@ -15,57 +15,66 @@ type ReplacementTracker interface {
 	Initialize(replacements map[string]string)
 
 	// Report returns a mapping of strings which were replaced.
-	Report() map[string]string
+	Report() ReplacementReport
 
 	// AddReplacement will add a replacement along with its original string to the report.
 	// If there is an existing value that does not match the given replacement, it will exit with a non-zero status.
-	AddReplacement(original string, replacement string)
+	AddReplacement(canonical, original string, replacement string)
```

Comment:
please also explain in the comment what canonical is and what difference there is to original